src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -218,7 +218,7 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, lon
                 bool goAway;
                 lock (SyncObj)
                 {
-                    goAway = _lastProcessedStreamId != -1 && requestStream.StreamId > _lastProcessedStreamId;
+                    goAway = _firstRejectedStreamId != -1 && requestStream.StreamId > _firstRejectedStreamId;
```

Comment:
I guess it should be >= now
```suggestion
                    goAway = _firstRejectedStreamId != -1 && requestStream.StreamId >= _firstRejectedStreamId;
```